lua/nvim-treesitter/parsers.lua

```diff
@@ -320,17 +307,15 @@ list.dhall = {
 list.diff = {
   install_info = {
     url = "https://github.com/the-mikedavis/tree-sitter-diff",
-    branch = "main",
```

Comment:
deleting branch `main` might cause problems when no lockfile is used because we will then assume `master`. We're not using `git`, so we need to decide for one branch. We can anything commitish via curl but I don't think there is anything like default-branch when no lockfile is used. Will this work with `branch` deleted? https://github.com/nvim-treesitter/nvim-treesitter/blob/350e7e5e31a5265fbccdf96b00a413c32583c2cc/lua/nvim-treesitter/install.lua#L691-L700 I think yes, it looks like `git ls-remote` gives the default branch first (though this might be just the first branch created. So it would have been incorrect even before)

Since I think nobody used nvim-treesitter with no lockfile we can do this clean up. Can we then remove the option to run without lockfile?

Reply:
> Can we then remove the option to run without lockfile?

I didn't know that was a thing...but yes as I wouldn't see why anyone wouldn't use the lockfile

Reply:
grep for `get_update_strategy`. In theory, you could always install from parser's master.

Reply:
> I didn't know that was a thing...but yes as I wouldn't see why anyone wouldn't use the lockfile

I don't use the lockfile when adding new languages. :)

Reply:
when adding locally or when adding them in a PR?

Reply:
Locally, before submitting the PR.

Reply:
Hmm me too, so what's the issue? or does branch need to be specified in that case as well if it's not master?

Reply:
I remember having to specify it.